inc/saibridge.h

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc
+ *
+ * @file    saibridge.h
+ *
+ * @brief   This module defines SAI Bridge
+ */
+
+#if !defined (__SAIBRIDGE_H_)
+#define __SAIBRIDGE_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIBRIDGE SAI - Bridge specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute data for #SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE
+ */
+typedef enum _sai_bridge_port_fdb_learning_mode_t
+{
+    /** Drop packets with unknown source MAC. Do not learn. Do not forward */
+    SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DROP,
+
+    /** Do not learn unknown source MAC. Forward based on destination MAC */
+    SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DISABLE,
+
+    /** Hardware learning. Learn source MAC. Forward based on destination MAC */
+    SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW,
+
+    /** Trap packets with unknown source MAC to CPU. Do not learn. Do not forward */
+    SAI_BRIDGE_PORT_FDB_LEARNING_MODE_CPU_TRAP,
+
+    /** Trap packets with unknown source MAC to CPU. Do not learn. Forward based on destination MAC */
+    SAI_BRIDGE_PORT_FDB_LEARNING_MODE_CPU_LOG,
+
+    /** Notify unknown source MAC using FDB callback. Do not learn in hardware. Do not forward.
+    * When a packet from unknown source MAC comes this mode will trigger a new learn notification
+    * via FDB callback for the MAC address. This mode will generate only one notification
+    * per unknown source MAC to FDB callback.
+    */
+    SAI_BRIDGE_PORT_FDB_LEARNING_MODE_FDB_NOTIFICATION,
+
+} sai_bridge_port_fdb_learning_mode_t;
+
+/**
+ * @brief Attribute data for #SAI_BRIDGE_PORT_ATTR_TYPE
+ */
+typedef enum _sai_bridge_port_type_t
+{
```

Comment:
currently we don't have admin state for every logical  interface in SAI E.g router interface  
I think that bridge port interface is not different that router interface
i think we should have this discussion but it is not mandatory for 1.0 
we can always add Admin state attribute leter do you agree ?